src/DocumentDB.ChangeFeedProcessor/Monitoring/HealthEventPhase.cs

```diff
@@ -0,0 +1,17 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+//----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.Monitoring
+{
+    /// <summary>
+    /// The health monitoring phase
+    /// </summary>
+    public enum HealthEventPhase
```

Comment:
I don't really like this name. Essentially, this is "the state the monitored system was in when health issue happened, or or operation/action of monitored system that triggered health issue". Also I think phase is too generic. So, how about MonitoredOperation (or TriggeringOperation [or Action] though I like 1st one better)?